### PR TITLE
Some minor refactoring on migration code

### PR DESCRIPTION
**Pull request by ppatierno**
Trivial PR to apply some minor refactoring to a couple of places related to the migration code.

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManager.java [hunk 1]
```diff
@@ -66,11 +67,7 @@ public KafkaMetadataStateManager(
         this.kraftAnno = kraftAnnotation(kafkaCr);
         KafkaMetadataState metadataStateFromKafkaCr = kafkaCr.getStatus() != null ? kafkaCr.getStatus().getKafkaMetadataState() : null;
         // missing metadata state means reconciling an already existing Kafka resource with newer operator supporting metadata state or first reconcile
-        if (metadataStateFromKafkaCr == null) {
-            this.metadataState = isKRaftAnnoEnabled() ? KafkaMetadataState.KRaft : KafkaMetadataState.ZooKeeper;
-        } else {
-            this.metadataState = metadataStateFromKafkaCr;
-        }
+        this.metadataState = Objects.requireNonNullElseGet(metadataStateFromKafkaCr, () -> isKRaftAnnoEnabled() ? KafkaMetadataState.KRaft : KafkaMetadataState.ZooKeeper);
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManager.java:70]: Why this change? Wasn't the original code easier to read while doing the same?

Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManager.java:70]: I see this code shorter for a simple thing and understandable at the same time like the ternary operator.

Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManager.java:70]: @strimzi/maintainers any more opinions on this?

Comment (see-quick) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManager.java:70]: Well, the old one was more readable because we are used to this I think :) the new one is just a fancy one-liner...I don't have a strong opinion about what is better...but for me, the readability is better in the old one but because I am more used to that form...

Comment (im-konge) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManager.java:70]: I agree that we are not much (I found one occurrence in STs) `requireNonNullElseGet` in the project and the previous format was less confusing FMPOV (as mentioned by @see-quick , just because we are not using it regularly).

Also, if you don't want to have the `else` block there, you can do
```
this.metadataState = metadataStateFromKafkaCr;

if (metadataStateFromKafkaCr == null) {
    this.metadataState = isKRaftAnnoEnabled() ? KafkaMetadataState.KRaft : KafkaMetadataState.ZooKeeper;
}
```
which will be even more confusing 😄 but it's an option.

Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManager.java]: ok, I reverted back to the previous code.

**Review by im-konge — APPROVED** [review]

**Review by see-quick — APPROVED** [review]

**Review by scholzj — APPROVED** [review]
